### PR TITLE
[SPARK-41579][SQL] Assign name to _LEGACY_ERROR_TEMP_1249

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_1249, "NOT_A_PARTITIONED_TABLE".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*`

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 1]
```diff
@@ -2405,22 +2405,24 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
...
-  def cmdOnlyWorksOnPartitionedTablesError(cmd: String, tableIdentWithDB: String): Throwable = {
+  def cmdOnlyWorksOnPartitionedTablesError(
+      operation: String,
+      tableIdentWithDB: String): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1249",
+      errorClass = "NOT_A_PARTITIONED_TABLE",
       messageParameters = Map(
-        "cmd" -> cmd,
+        "operation" -> operation,
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2414]: Please, quote it using `toSQLStmt()`.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — CHANGES_REQUESTED** [review]
Please, add a test for the error class.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 2]
```diff
@@ -2405,22 +2405,24 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
...
-      errorClass = "_LEGACY_ERROR_TEMP_1249",
+      errorClass = "NOT_A_PARTITIONED_TABLE",
       messageParameters = Map(
-        "cmd" -> cmd,
+        "operation" -> toSQLStmt(operation),
         "tableIdentWithDB" -> tableIdentWithDB))
   }
 
-  def cmdOnlyWorksOnTableWithLocationError(cmd: String, tableIdentWithDB: String): Throwable = {
+  def cmdOnlyWorksOnTableWithLocationError(
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:2418]: This is another error, isn't it? It is about location not about partitioning. Please, fix this.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 3]
```diff
@@ -2166,6 +2166,22 @@ abstract class DDLSuite extends QueryTest with DDLSuiteBase {
       assert(spark.sessionState.catalog.isRegisteredFunction(rand))
     }
   }
+
+  test("NOT_A_PARTITIONED_TABLE: ALTER TABLE is not allowed for non-partitioned table") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala:2170]: Precisely speaking,  `ALTER TABLE .. RECOVER PARTITIONS` not just `ALTER TABLE`

**Review by MaxGekk — CHANGES_REQUESTED** [review]
There are dedicated test suites `AlterTableRecoverPartitionsSuite` for the command, could you place the test there, please.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala (hunk 3 above)
Comment (itholic) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: Just moved the test into dedicated test suites, and update the test name.
Thanks!

File: core/src/main/resources/error/error-classes.json [hunk 4]
```diff
@@ -961,6 +961,11 @@
     ],
     "sqlState" : "42000"
   },
+  "NOT_A_PARTITIONED_TABLE" : {
+    "message" : [
+      "Operation <operation> is not allowed for <tableIdentWithDB> because it is not a partitioned table."
```
Comment (cloud-fan) [inline core/src/main/resources/error/error-classes.json:966]: `qualifiedTableName` is better than `tableIdentWithDB` but it's minor.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

File: core/src/main/resources/error/error-classes.json (hunk 4 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json:966]: Let me handle this when I submit another error message improvement PR.
Thanks!

**Pull request closed without merging**